src/proxy_handler.cpp

```diff
@@ -0,0 +1,206 @@
+//
+// Created by Michael Chen on 2/28/17.
+// Copyright © 2017 Chen. All rights reserved.
+//
+//ProxyHandler code for generating new request and capturing response is modeled off of
+//http://www.boost.org/doc/libs/1_55_0/doc/html/boost_asio/example/cpp03/http/client/sync_client.cpp
+
+#include "proxy_handler.h"
+#include "parsed_config.h"
+#include <istream>
+#include <ostream>
+#include <string>
+#include <vector>
+#include <boost/bind.hpp>
+#include <boost/asio.hpp>
+#include "utils.h"
+#include <boost/regex.hpp>
+
+
+RequestHandler::Status ProxyHandler::Init(const std::string& uri_prefix,
+					 const NginxConfig& config) {
+  uri_prefix_ = uri_prefix;
+  ParsedConfig parser;
+  proxy_host_ = parser.GetStatementValue(&config, "host");
+  if (proxy_host_.length() == 0) {
+    return RequestHandler::INVALID_CONFIG;
+  }
+  
+  proxy_port_ = parser.GetStatementValue(&config, "port");
+  if (proxy_port_.length() == 0) {
+    return RequestHandler::INVALID_CONFIG;
+  }
+
+  redirect_uri_ = "";
+  return RequestHandler::OK;
+}
+
+RequestHandler::Status ProxyHandler::HandleRequest(const Request &request, 
+                                                   Response *response) {
+  return HandleRequestHelper(request, response, 0);
+}
+
+RequestHandler::Status ProxyHandler::HandleRequestHelper(const Request& request,
```

Comment:
This function is way too long. Split it up into a couple of functions - one that sends the initial proxy request, another that reads the response, and one more that modifies the URLs in the HTML.